t/t2355-resource-journal.t

```diff
@@ -153,7 +153,44 @@ test_expect_success 'last event: resource-define method=kvs' '
 	jq -e ".name == \"resource-define\"" restartlog.3 &&
 	jq -e ".context.method == \"kvs\"" restartlog.3
 '
-
+test_expect_success 'ensure all ranks are undrained' '
+	ranks=$(flux resource status -no {ranks} -s drain) &&
+	if test -n "$ranks"; then
+	    flux resource undrain $ranks
+	fi
+'
+test_expect_success 'capture eventlog before truncation' '
+	flux resource eventlog -H &&
+	flux resource eventlog -f json > eventlog.pre.out
+'
+test_expect_success 'set a journal size limit 1 less than current entries' '
+	limit=$(($(flux resource eventlog | wc -l) - 1)) &&
+	echo resource.journal-max=$limit | flux config load
+'
+test_expect_success 'eventlog is now truncated' '
+	flux resource eventlog -H &&
+	flux resource eventlog -f json > eventlog.trunc
+'
+test_expect_success '1st event is a truncate event' '
+	head -1 eventlog.trunc > eventlog.trunc.1 &&
+	jq -e ".name == \"truncate\"" eventlog.trunc.1
+'
```

Comment:
suggestion: also check number of entries == limit, and likewise with 2 tests below this one